plugins/woocommerce-blocks/tests/e2e/tests/product-collection/register-product-collection-tester.block_theme.spec.ts

```diff
@@ -357,3 +357,111 @@ test.describe( 'Product Collection registration', () => {
 		} );
 	} );
 } );
+
+test.describe( 'Deleted product reference in Product Collection block', () => {
+	let testProductId: number | null = null;
+
+	test.beforeEach( async ( { requestUtils } ) => {
+		// Activate plugin
+		await requestUtils.activatePlugin(
+			'register-product-collection-tester'
+		);
+
+		// Check if "A Test Product" exists
+		const products = await requestUtils.rest( {
+			method: 'GET',
+			path: 'wc/v3/products',
+			params: {
+				search: 'A Test Product',
+			},
+		} );
+
+		// Add "Test Product" only if it doesn't exist
+		if ( products.length === 0 ) {
+			const newProduct = await requestUtils.rest( {
+				method: 'POST',
+				path: 'wc/v3/products',
+				data: {
+					name: 'A Test Product',
+					price: 10,
+				},
+			} );
+			testProductId = newProduct.id;
+		} else {
+			testProductId = products[ 0 ].id;
+		}
+	} );
+
+	test( 'Product picker should be shown when product reference is deleted', async ( {
+		pageObject,
+		admin,
+		editor,
+		requestUtils,
+	} ) => {
+		await admin.createNewPost();
+		await pageObject.insertProductCollection();
+		await pageObject.chooseCollectionInPost(
+			'myCustomCollectionWithProductContext'
+		);
+
+		// Verify that product picker is shown in Editor
+		const editorProductPicker = editor.canvas.locator(
+			SELECTORS.productPicker
+		);
+		await expect( editorProductPicker ).toBeVisible();
+
+		// Once a product is selected, the product picker should be hidden
+		await pageObject.chooseProductInEditorProductPickerIfAvailable(
+			editor.canvas,
+			'A Test Product'
+		);
+		await expect( editorProductPicker ).toBeHidden();
+
+		await editor.saveDraft();
+
+		// Delete the product
+		if ( testProductId ) {
+			await requestUtils.rest( {
+				method: 'DELETE',
+				path: `wc/v3/products/${ testProductId }`,
+			} );
+		}
```

Comment:
Same here, we shouldn't need to check if the product exists.

Reply:
Aha! Good catch. I have made the changes in [6c8ab82](https://github.com/woocommerce/woocommerce/pull/51114/commits/6c8ab8266ccb6ec60cafff2bdad8da2c3e71b7d7)